mindmeld/components/dialogue.py

```diff
@@ -689,6 +691,187 @@ def _get_dialogue_handler(self, dialogue_state):
         return handler
 
 
+class AutoEntityFilling(DialogueFlow):
+    """A special dialogue flow sublcass to implement Automatic Entitiy (Slot) Filling
+    (AEF) that allows developers to prompt users for completing the missing
+    requirements for entity slots.
+
+    Attributes:
+        app (Application): The application that initializes this flow.        
+    """
+
+    logger = mod_logger.getChild("AutoEntityFilling")
+    """Class logger."""
+
+    def __init__(self, name, entrance_handler, app, **kwargs):
+        super().__init__(name, entrance_handler, app)
+
+        self.app = app
+        self.entrance_handler = entrance_handler
+
+    def _set_target_state(self):
+        """Set target dialogue state to current flow"""
+        responder.params.target_dialogue_state = self.flow_state
+
+    def _extract_query_features(self, text):
+        """ Extracts Query object from the user input and converts it into
+        appropriate format for entity extraction.
+
+        Args:
+            text (Request.text): Text in the query
+
+        Returns:
+            Formatted Payload (tuple(query (Query Object), list of entities, entity index))
+        """
+        resource_loader = self.app.app_manager.nlp.resource_loader
+
+        """ Extracting the Query object associated with the user input here.
+
+        process_markup args:
+            text, query factory, query options (optional dict) {language, time_zone and time_stamp}
+
+        returns: 
+            tuple(raw_text (str), Query object, list of entities)
+        """
+        query = process_markup(text, resource_loader.query_factory, {})[1]
+
+
+        """ 
+        payload format for entity feature extractors: 
+            tuple(query (Query Object), list of entities, entity index)
+
+        For slot filling, the user query will consist of the prompted entity. 
+        Hence `entity = [query]` and `entity_index = 0`.
+
+        """
+        formatted_payload = (query, [query], 0)
+        return formatted_payload
```

Comment:
We shouldn't need to re-run this right? Instead pass in the extracted entities from the request object?